MonoGame.Framework.Content.Pipeline/Processors/EffectProcessor.cs

```diff
@@ -50,45 +50,112 @@ public EffectProcessor()
         public override CompiledEffectContent Process(EffectContent input, ContentProcessorContext context)
         {
             var mgfxc = Path.Combine(Path.GetDirectoryName(typeof(EffectProcessor).Assembly.Location), "mgfxc.dll");
-            var sourceFile = Path.GetTempFileName();
+            var sourceFile = input.Identity.SourceFilename;
             var destFile = Path.GetTempFileName();
-            var target = context.TargetPlatform.ToString().Contains("Windows");
-            
-            File.WriteAllText(sourceFile, input.EffectCode);
-
-            var proc = new Process();
-            proc.StartInfo.FileName = "dotnet";
-            proc.StartInfo.Arguments = "\"" + mgfxc + "\" \"" + sourceFile + "\" \"" + destFile + "\" /Profile:" + GetProfileForPlatform(context.TargetPlatform.ToString());
+            var arguments = "\"" + mgfxc + "\" \"" + sourceFile + "\" \"" + destFile + "\" /Profile:" + GetProfileForPlatform(context.TargetPlatform);
 
             if (debugMode == EffectProcessorDebugMode.Debug)
-                proc.StartInfo.Arguments += " /Debug";
+                arguments += " /Debug";
 
             if (!string.IsNullOrWhiteSpace(defines))
-                proc.StartInfo.Arguments += " \"/Defines:" + defines + "\"";
+                arguments += " \"/Defines:" + defines + "\"";
 
-            proc.Start();
-            proc.WaitForExit();
+            string stdout, stderr;
 
-            var success = proc.ExitCode == 0;
+            var success = ExternalTool.Run("dotnet", arguments, out stdout, out stderr) == 0;
             var ret = success ? new CompiledEffectContent(File.ReadAllBytes(destFile)) : null;
 
-            File.Delete(sourceFile);
             File.Delete(destFile);
 
+            ProcessErrorsAndWarnings(!success, stderr, input, context);
+
             if (!success)
```

Comment:
@harry-cpp - this block can be removed.  `ProcessErrorsAndWarnings` handles the failure case.

Reply:
That is the only position where `ProcessErrorsAndWarnings` is getting called from. I've also modified mgfxc to output warnings and errors to stderr, while other user related information to stdout.